cocos/audio/android/AudioEngine-inl.cpp

```diff
@@ -280,6 +280,10 @@ int AudioEngineImpl::play2d(const std::string &filePath ,bool loop ,float volume
                 {
                     _audioPlayers.erase(id);
                 }
+                if (_urlAudioPlayersNeedResume.find(id) != _urlAudioPlayersNeedResume.end())
+                {
+                    _urlAudioPlayersNeedResume.erase(id);
```

Comment:
Therefore, the function invocation order may be:
* onEnterBackground: Save UrlAudioPlayer to map.
* PlAY OVER event comes: but the old implementation didn't remove it from vector.
* onEnterForeground: Resume all saved `UrlAudioPlayer`s, since PLAY OVER has come, it has already been deleted, `UrlAudioPlayer` in vector will be an invalid pointer and any invocation on that pointer will cause crash.

Is the above logic represents the issue?